File: 387. First Unique Character in a String/main.py
import collections
class Solution:
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        d=collections.OrderedDict()
        l=len(s)
        for i in range(l):
            if s[i] not in d.keys():
                d[s[i]]=i
            else:
                d[s[i]]=-1
                
        for k in d.keys():
            if d[k]!=-1:
                return d[k]
            
        return -1

File: 387. First Unique Character in a String/main.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int l=s.length(),i;
        map<char,int> m;
        
        for(i=0;i<l;i++){
            if(m.count(s[i])==0)
                m[s[i]]=i;
            else
                m[s[i]]=-1;
        }
        for(i=0;i<l;i++){
            if(m[s[i]]!=-1)
                return m[s[i]];
        }
        return -1;
    }
};
